components/controller.ts

```diff
@@ -831,6 +840,26 @@ export class Controller {
         }
     }
 
+    private async _loadMasteryData(): Promise<void> {
```

Comment:
This should probably be combined with `_loadChallenges` into a `_loadResources` function.

Reply:
See comment above. I'm all for going for a generic resource package format. If this were to become loadResources that makes sense. However, from a Single Responsibility point of view while it does load resources, it also does the registration for the specific type of the data it found. I would rather go for a _loadResources-function that does the byte reading, and then two functions (as callbacks?) to handle the actual processing.